### PR TITLE
[ODS-5424] Update Tests to run against Postgres

### PR DESCRIPTION
**Pull request by ericstockdale**

### CODE REVIEW

**Review by bptillman — CHANGES_REQUESTED** [review]
Great job on this by the way. It was a lot of work. A lot more than we/I expected. The only issues I've made comments on are all minor and nothing structurally I can see that needs to change.

File: Application/EdFi.Ods.Repositories.NHibernate.Tests/BaseDatabaseTest.cs [hunk 1]
```diff
@@ -13,8 +12,11 @@
 using EdFi.Ods.Repositories.NHibernate.Tests.Modules;
 using Microsoft.Extensions.Configuration;
 using NHibernate;
-using NHibernate.Cfg;
 using NUnit.Framework;
+using System.Reflection;
+using Environment = NHibernate.Cfg.Environment;
+using PostgresSpecificModule = EdFi.Ods.Repositories.NHibernate.Tests.Modules.PostgresSpecificModule;
```
[the remark this answers is not in the document]
Comment (ericstockdale) [reply Application/EdFi.Ods.Repositories.NHibernate.Tests/BaseDatabaseTest.cs]: Removed

File: Application/EdFi.Ods.Repositories.NHibernate.Tests/StudentProgramAssociationTests.cs [hunk 2]
```diff
@@ -175,12 +176,13 @@ protected override void Act()
                     IsModified = studentProgramAssociationUpsertResult.IsModified
                 };
 
+                var databaseEngine = DbHelper.GetDatabaseEngine();
                 // Verify the service got removed
                 using (var conn = GetSqlConnectionForOds())
```
[the remark this answers is not in the document]
Comment (ericstockdale) [reply Application/EdFi.Ods.Repositories.NHibernate.Tests/StudentProgramAssociationTests.cs]: Renamed

File: Application/EdFi.Ods.Repositories.NHibernate.Tests/StudentProgramAssociationTests.cs [hunk 3]
```diff
@@ -33,8 +26,15 @@
...
 using NUnit.Framework;
 using Shouldly;
+using System;
+using System.Data;
+using System.Reflection;
+using System.Security.Claims;
+using System.Threading;
 using Test.Common.DataConstants;
-
+using PostgresSpecificModule = EdFi.Ods.Repositories.NHibernate.Tests.Modules.PostgresSpecificModule;
```
[the remark this answers is not in the document]
Comment (ericstockdale) [reply Application/EdFi.Ods.Repositories.NHibernate.Tests/StudentProgramAssociationTests.cs]: Removed

File: tests/EdFi.Admin.DataAccess.IntegrationTests/appsettings.json [hunk 4]
```diff
@@ -1,5 +1,3 @@
 {
-  "ConnectionStrings": {
-    "EdFi_Admin": "Database=EdFi_Admin_Test; Data Source=(local); Trusted_Connection=True;MultipleActiveResultSets=True"
-  }
+  "Engine": "PostgreSql"
```
[the remark this answers is not in the document]
Comment (ericstockdale) [reply tests/EdFi.Admin.DataAccess.IntegrationTests/appsettings.json]: Added comment to all settings.json and also made sure they were all sqlServer by default

File: tests/EdFi.Ods.WebApi.IntegrationTests/DbHelper.cs [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+using Npgsql;
+using NUnit.Framework;
+using System.Data;
+using System.Data.SqlClient;
+
+namespace EdFi.Ods.WebApi.IntegrationTests
+{
+    public class DbHelper
+    {
+        public static IConfigurationRoot GetDatabaseEngineSpecificConfiguration(DatabaseEngine databaseEngine)
```
[the remark this answers is not in the document]
Comment (ericstockdale) [reply tests/EdFi.Ods.WebApi.IntegrationTests/DbHelper.cs:17]: Removed.  The other integration tests used this method.  However, WebApi.IntegrationTests uses a Host that runs the configuration builder so this method is inapplicable for this project.

File: tests/EdFi.Ods.WebApi.CompositeSpecFlowTests/EnvironmentSetupTests.cs [hunk 6]
```diff
@@ -3,13 +3,16 @@
 // The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
 // See the LICENSE and NOTICES files in the project root for more information.
 
+using System.Data;
```
[the remark this answers is not in the document]
Comment (ericstockdale) [reply tests/EdFi.Ods.WebApi.CompositeSpecFlowTests/EnvironmentSetupTests.cs]: removed

File: tests/EdFi.Ods.WebApi.CompositeSpecFlowTests/Steps.cs [hunk 7]
```diff
@@ -69,13 +80,29 @@ public async Task GivenTheSubjectOfTheRequestIsAStudentEducationOrganizationAsso
         [Given(@"the subject of the request is a school with student and staff associations")]
         public async Task GivenTheSubjectOfTheRequestIsASchoolWithStudentAndStaffAssociations()
         {
-            const string query = @"
+            string query = string.Empty;
```
[the remark this answers is not in the document]
Comment (ericstockdale) [reply tests/EdFi.Ods.WebApi.CompositeSpecFlowTests/Steps.cs]: removed

**Review by bptillman — APPROVED** [review]

**Pull request merged**